lib/api/kuzzle.js

```diff
@@ -32,72 +53,80 @@ const
 
 /**
  * @class Kuzzle
- * @constructor
+ * @extends EventEmitter
  */
-function Kuzzle () {
-  /** @type {KuzzleConfiguration} */
-  this.config = require('../config');
-  this.config.version = packageInfo.version;
+class Kuzzle extends EventEmitter {
+  constructor() {
+    super({
+      wildcard: true,
+      maxListeners: 30,
+      delimiter: ':'
+    });
 
-  this.rootPath = path.resolve(path.join(__dirname, '..', '..'));
+    /** @type {KuzzleConfiguration} */
+    this.config = require('../config');
+    this.config.version = packageInfo.version;
 
-  this.services = new Services(this);
-  this.cli = new Cli(this);
-  this.internalEngine = new InternalEngine(this);
-  this.pluginsManager = new PluginsManager(this);
-  this.tokenManager = new TokenManager(this);
-  this.indexCache = new IndexCache(this);
-  this.repositories = new Repositories(this);
+    this.rootPath = path.resolve(path.join(__dirname, '..', '..'));
 
-  this.gc = new GarbageCollector(this);
+    this.services = new Services(this);
+    this.cli = new Cli(this);
+    this.internalEngine = new InternalEngine(this);
+    this.pluginsManager = new PluginsManager(this);
+    this.tokenManager = new TokenManager(this);
+    this.indexCache = new IndexCache(this);
+    this.repositories = new Repositories(this);
 
-  this.passport = new PassportWrapper();
+    this.gc = new GarbageCollector(this);
 
-  // The funnel controller dispatch messages between the router controller and other controllers
-  this.funnel = new FunnelController(this);
+    this.passport = new PassportWrapper();
 
-  // The router controller listens to client requests and pass them to the funnel controller
-  this.router = new RouterController(this);
+    // The funnel controller dispatch messages between the router controller and other controllers
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 98 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)